aiohttp/helpers.py

```diff
@@ -297,6 +297,20 @@ def isasyncgenfunction(obj: Any) -> bool:
         return False
 
 
+def get_env_proxy_for_url(url: URL) -> Tuple[URL, Optional[BasicAuth]]:
+    """Get a permitted proxy for the given URL from the env."""
+    if url.host is not None and proxy_bypass(url.host):
+        raise LookupError(f"Proxying is disallowed for `{url.host!r}`")
+
+    proxies_in_env = proxies_from_env()
+    try:
+        proxy_info = proxies_in_env[url.scheme]
+    except KeyError:
+        raise LookupError(f"No proxies found for `{url!s}` in the env")
+    else:
+        return proxy_info.proxy, proxy_info.proxy_auth
+
+
 @attr.s(auto_attribs=True, frozen=True, slots=True)
```

Comment:
The merge add a `@dataclasses.dataclass(frozen=True)` here. But since I didn't see it used anywhere else in this file in the 3.8 code I removed it.